CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -23,35 +25,70 @@ local topbarEnabled = false
 local player = game.Players.LocalPlayer
 local currentHumanoid = nil
 local CoreGui = game:GetService('CoreGui')
+local RobloxGui = CoreGui:WaitForChild("RobloxGui")
+local Panel3D = require(RobloxGui.Modules.Panel3D)
 
-local hopperbinPart = Instance.new("Part", workspace.CurrentCamera)
-hopperbinPart.Transparency = 1
-hopperbinPart.CanCollide = false
-hopperbinPart.Anchored = true
-hopperbinPart.Name = "GUI"
-local hopperbinGUI = Instance.new("SurfaceGui", CoreGui)
-hopperbinGUI.Adornee = hopperbinPart
-hopperbinGUI.ToolPunchThroughDistance = 1000
-hopperbinGUI.Name = "HopperBinGUI"
-local toolsFrame = Instance.new("Frame", hopperbinGUI)
+local ContextActionService = game:GetService("ContextActionService")
+
+local panel = Panel3D.Get(Panel3D.Panels.Lower)
+local toolsFrame = Instance.new("TextButton", panel.gui) --prevent clicks falling through in case you have a rocket launcher and blow yourself up
```

Comment:
We should make this a Frame with the `Active` property set to true. That way it will still sink clicks and we don't have a strange TextButton floating around